sabnzbd/api.py

```diff
@@ -1317,15 +1317,39 @@ def test_nntp_server_dict(kwargs: Dict[str, Union[str, List[str]]]) -> Tuple[boo
     # All exceptions are caught internally
     test_server.request_addrinfo_blocking()
     if not test_server.addrinfo:
-        # Try if we can connect on port 80 (so web server), forcing a short timeout
+        # so NNTP connection was tried, but did not succeed: no addrinfo. Possible causes:
+        # - user has filled out an indexer as newsserver. Not good.
+        # - user has filled out a weird port on which there is no newsserver
+        # - generic network problem (?)
+
+        test_server.timeout = DEF_NETWORKING_TEST_TIMEOUT  # force a short timeout
+        # let's try well-known ports: HTTP and NTTP(S)
         test_server.port = 80
-        test_server.timeout = DEF_NETWORKING_TEST_TIMEOUT
         test_server.request_addrinfo_blocking()
-        if test_server.addrinfo:
+        port80working = bool(test_server.addrinfo)
+
+        test_server.port = 119  # NNTP
```

Comment:
We should only try 119 if the port we originally tested wasn't 119. Otherwise we are testing twice. Yes, I know the HappyEyeBalls has caching, so could also just note that in a comment.
Same for 563.

Reply:
I thought this simple logic was better ... but I'll change.